src/ring_theory/ideal_operations.lean

```diff
@@ -547,14 +547,15 @@ def ker : ideal R := ideal.comap f ⊥
 
 lemma ker_eq : ((ker f) : set R) = is_add_group_hom.ker f := rfl
 
-lemma inj_of_trivial_ker (h : ker f = ⊥) : function.injective f :=
-is_add_group_hom.inj_of_trivial_ker f $ by rw [←ker_eq, h]; refl
+lemma inj_iff_ker_eq_bot : function.injective f ↔ ker f = ⊥ :=
+by rw [←submodule.ext'_iff, ker_eq]; exact is_add_group_hom.inj_iff_trivial_ker f
 
-lemma trivial_ker_of_inj (h : function.injective f) : ker f = ⊥ :=
-submodule.ext' $ by rw [ker_eq, is_add_group_hom.trivial_ker_of_inj f h]; refl
+lemma ker_eq_bot_iff_eq_zero : ker f = ⊥ ↔ ∀ x, f x = 0 → x = 0 :=
+by rw [←submodule.ext'_iff, set.ext_iff]; simp [ker]; exact
+⟨λ h x hx, (h x).1 hx, λ h x, ⟨h x, λ hx, by rw [hx, is_ring_hom.map_zero f]⟩⟩
 
-lemma inj_iff_trivial_ker : function.injective f ↔ ker f = ⊥ :=
-⟨trivial_ker_of_inj f, inj_of_trivial_ker f⟩
+lemma inj_iff_eq_zero : function.injective f ↔ ∀ x, f x = 0 → x = 0 :=
```

Comment:
This lemma is not specific to ring homs. So it should be proven for group homs (and additive group homs, using `to_additive`). It should of course go in the correct file `group_theory/subgroup`. After that, this looks ready for merging, I think.